storey/targets.py

```diff
@@ -629,7 +633,12 @@ async def _emit(self, batch, batch_key, batch_time, batch_events, last_event_tim
             dir_path += "/"
         if dir_path and not self._file_system.exists(dir_path):
             self._file_system.makedirs(dir_path, exist_ok=True)
-        file_path = self._path if self._single_file_mode else f"{dir_path}{uuid.uuid4()}.parquet"
+        if not self._single_file_mode:
+            file_path = f"{dir_path}{uuid.uuid4()}.parquet"
+        elif self._single_file_mode and not self._partition_cols:
+            file_path = self._path
+        else:
+            file_path = f"{dir_path}{self.name}.parquet"
```

Comment:
Logic could use a bit of refactoring here:
```suggestion
        if self._single_file_mode:
            file_path = f"{dir_path}{self.name}.parquet" if self._partition_cols else self._path
        else:
            file_path = f"{dir_path}{uuid.uuid4()}.parquet"
```